src/Table.cs

```diff
@@ -28,22 +27,50 @@ public interface IRemoteTableHandle
         internal string RemoteTableName { get; }
 
         internal Type ClientTableType { get; }
-        internal IStructuralReadWrite DecodeValue(BinaryReader reader);
+
+        /// <summary>
+        /// Creates and returns a parsed table update for the current table.
+        /// </summary>
+        /// <returns>An <see cref="IParsedTableUpdate"/> representing the parsed update.</returns>
```

Comment:
Note about type erasure here: To use an IParsedTableUpdate, you have to downcast it to its particular concrete type. This interface just needs to be type erased because `IRemoteTableHandle` is type erased.